### PR TITLE
AARCH64 assembly support

### PR DESCRIPTION
**Pull request by troy4eg**
- improved macos (m processor series) support
- linux on arm processors support

### CODE REVIEW

File: cmake/init-compilation-flags.cmake [hunk 1]
```diff
@@ -80,6 +80,9 @@ add_compile_options(-fwrapv -fno-strict-aliasing -fno-stack-protector -ggdb -fno
 if(CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")
     add_compile_options(-march=sandybridge -fno-common)
     add_link_options(-fno-common)
+elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "aarch64")
+    add_compile_options(-march=armv8.2-a+crypto+fp16+rcpc+dotprod+ssbs -fno-common -fsigned-char)
```
Comment (artem) [inline cmake/init-compilation-flags.cmake:84]: Maybe replace -march flag with `-mcpu=neoverse-n1+crypto`?

File: compiler/compiler-settings.cpp [hunk 2]
```diff
@@ -281,6 +284,8 @@ void CompilerSettings::init() {
   ss << " -fno-strict-aliasing -fno-omit-frame-pointer";
 #ifdef __x86_64__
   ss << " -march=sandybridge";
+#elif __aarch64__
+  ss << " -march=armv8.2-a+crypto+fp16+rcpc+dotprod+ssbs";
```
Comment (artem) [inline compiler/compiler-settings.cpp:288]: ditto

File: cmake/init-compilation-flags.cmake [hunk 3]
```diff
@@ -88,10 +88,12 @@ if(NOT DEFINED ENV{ENABLE_GRPROF})
...
 include_directories(${GENERATED_DIR})
-add_compile_options(-fwrapv -fno-strict-aliasing -fno-stack-protector -ggdb -fno-omit-frame-pointer)
+add_compile_options(-fwrapv -fno-strict-aliasing -fno-stack-protector -ggdb -fno-omit-frame-pointer -fno-common -fsigned-char)
+add_link_options(-fno-common)
 if(CMAKE_SYSTEM_PROCESSOR STREQUAL "x86_64")
-    add_compile_options(-march=sandybridge -fno-common)
-    add_link_options(-fno-common)
+    add_compile_options(-march=sandybridge)
+elseif(CMAKE_SYSTEM_PROCESSOR STREQUAL "aarch64")
+    add_compile_options(-march=armv8.2-a+crypto+fp16+rcpc+dotprod+ssbs)
```
Comment (artem) [inline cmake/init-compilation-flags.cmake:96]: If we want to play it safe, let's limit it to `armv8.2-a+crypto` then

File: common/precise-time.h [hunk 4]
```diff
@@ -2,28 +2,25 @@
...
-extern long long precise_time;  // (long long) (2^16 * precise unixtime)
-extern long long precise_time_rdtsc; // when precise_time was obtained
-long long get_precise_time (unsigned precision);
+double get_precise_time(unsigned precision) noexcept;
 
-/* ??? */
-double get_double_time ();
+static inline long long precise_time_to_binlog_time(double precise_time) noexcept {
+  return static_cast<long long>(precise_time * (1LL << 32));
+}
```
Comment (DrDet) [inline common/precise-time.h:20]: Why do we need this? KPHP doesn't have binlog

File: common/tl/query-header.cpp [hunk 5]
```diff
@@ -382,7 +382,7 @@ void set_result_header_values(tl_query_answer_header_t *header, int flags) {
   }
   if (flags & request_flag::return_binlog_time) {
     header->flags |= result_flag::binlog_time;
-    header->binlog_time = get_precise_time(1000000);
+    header->binlog_time = precise_time_to_binlog_time(get_precise_time(1000000));
```
Comment (DrDet) [inline common/tl/query-header.cpp:385]: and here

**Review by DrDet — APPROVED** [review]

**Review by DrDet — DISMISSED** [review]

**Review by DrDet — APPROVED** [review]

**Pull request merged**